typed-racket-lib/typed-racket/typecheck/toplevel-trampoline.rkt

```diff
@@ -157,28 +158,37 @@
             ;; before printing them.
             (define tc (cleanup-type t))
             (define tg (generalize tc))
```

Comment:
Should these happen in the other order?

Reply:
`cleanup-type` is intended for function types and generalize doesn't deal with function types, so I don't think the order matters in this case.